### DISCUSSION TITLE
Upload fails with `Input image exceeds pixel limit`

### DISCUSSION THREAD

**Discussion by aeneasr** [argos-ci/argos#1140] [Q&A] [answered]
### Motivation

I am trying to upload screenshots to argos. However, running argos upload fails with:

```
$ npm exec argos upload ./screenshots
✖ Build failed: Input image exceeds pixel limit
Error: Input image exceeds pixel limit
```

### Detail

As there are no details about which input image exceeds the pixel limit, it is quite challenging to figure out where the problem is. It would be great it at least the image name was given.

Alternatively, it would be great if there ways to deal with big screenshots or know what the pixel limit is.

**Comment by jsfez** [comment]
Hi @aeneasr, 
It seems some images exceed the size limit set by the Sharp library, which Argos uses. The current limit is width x height = 268402689. 

You can identify oversized images by running tests locally. You are right, it would be better to have the list of failed uploads, but it's not a priority for now.

We haven't considered increasing this limit as there hasn't been a need. Could you share why larger images are necessary for your use case?

**Comment by aeneasr** [comment]
Hey, they are not neccessary for the use case and I'd gladly ignore them in the upload, but given that we have around 500 pages it's not possible to understand which image(s) are causing the issues.

**Comment by jsfez** [comment]
I understand your problem. I'll push a feature to help you. It should be done today or tomorrow

**Comment by gregberge** [comment]
Hello @aeneasr, we release a new version of `@argos-ci/cli` with an improved error display that gives you the path of the file that failed to be processed by sharp. Could you upgrade `@argos-ci/cli` and try to remove the file that causes problem, or tell us about it?

**Comment by gregberge** [comment, marked as answer]
The problem is now solved, it was an issue on the Argos CLI.
